crates/egui/src/containers/scroll_area.rs

```diff
@@ -751,6 +761,13 @@ impl Prepared {
             }
         }
 
+        // Restore scroll target (if any) meant for ScrollAreas up the stack
+        if saved_scroll_target[0].is_some() || saved_scroll_target[1].is_some() {
+            ui.ctx().frame_state_mut(|state| {
+                state.scroll_target = saved_scroll_target;
```

Comment:
If either the horizontal or the vertical part of `saved_scroll_target` is set, overwrite both.
If neither is set, overwrite neither.

Without having thought deeply about this, this seems wrong to me.

Why not always write to `state.scroll_target`?
Or why not do the logic per dimension?

Reply:
I changed it to be per dimension since it's more readable. I was trying to prevent locking twice but probably not worth it

Reply:
We could also _always_ restore the saved scroll target, a few extra cycles but not worth worrying about. Let me know and I can change it.